limits.c

```diff
@@ -125,42 +125,40 @@ void limits_go_home(uint8_t cycle_mask)
   // Determine travel distance to the furthest homing switch based on user max travel settings.
   float min_seek_rate = 1e9; //arbitrary maximum=1km/s, will be reduced by axis setting below
   float max_travel = 0;
+  uint8_t n_active_axis = 0;
+  uint8_t axislock = 0; 
 
   for (idx=0; idx<N_AXIS; idx++){
     if (bit_istrue(cycle_mask,bit(idx))) {
+      n_active_axis++;
+      axislock |= (1<<(X_STEP_BIT+idx)); //assumes axes are in bit order.
       max_travel = max(max_travel,settings.max_travel[idx]);
       min_seek_rate = min(min_seek_rate,settings.homing_seek_rate[idx]);
     }
   }
   max_travel *= HOMING_AXIS_SEARCH_SCALAR; // Ensure homing switches engaged by over-estimating max travel.
   max_travel += settings.homing_pulloff;
-  homing_rate = min_seek_rate;
+  homing_rate = min_seek_rate * sqrt(n_active_axis); //Adjust so individual axes all move at homing rate.
```

Comment:
This was just moved from L157.  But the answer is that the feed rate is the rate of a straight line move.  If we are moving two orthoginal axes at once, we want to move at sqrt(2) = 1.4x along the diagonal to get 1x speed along each axis.